olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionedLayerClient.h

```diff
@@ -500,6 +500,52 @@ class DATASERVICE_READ_API VersionedLayerClient final {
    */
   bool Release(const TileKeys& tiles);
 
+  /**
+   * @brief Protect partition from eviction.
+   *
+   * Protecting partition means that its data and metadata keys are added to the
+   * protected list and stored in the cache. These keys are removed from the LRU
+   * cache, so they could not be evicted. Also, they do not expire. The quadtree
+   * stays protected if at least one tile key is protected.
+   *
+   * @note You can only protect partitions which data handles are present in the
+   * cache at the time of the call.
+   *
+   * @note Please do not call `Protect` while the `Release` call for the same
+   * catalog and layer is in progress.
+   *
+   * @note Before calling the API, specify a layer version. You can set it using
+   * the constructor or after the first online request.
+   *
+   * @param partition_id Partition id to be protected.
+   *
+   * @return True if some keys were successfully added to the protected list;
+   * false otherwise.
+   */
+  bool Protect(const std::string& partition_id);
```

Comment:
I would group all `Protect` and all `Release` methods and not provide them in pairs.